homeassistant/components/media_player/__init__.py

```diff
@@ -106,6 +109,7 @@
 SUPPORT_CLEAR_PLAYLIST = 8192
 SUPPORT_PLAY = 16384
 SUPPORT_SHUFFLE_SET = 32768
+SUPPORT_SELECT_SOUND_MODE = 65536
```

Comment:
We should not add this as a new supported feature for media players. We should only add things that exist among many media players.

Revert all changes to this file and add a new service to the denonavr platform called `media_player.denonavr_set_sound_mode` to allow changing thet mode.

Reply:
I was planning to add support for sound modes in a form or another to songpal, so that'd be two platforms for using such a feature. I haven't checked out yet how to do that, but will try to find some time during the upcoming weekend for it, if that's fine?

Reply:
@balloob I asked about this on the form when I first started with this code, and at that time there was also someone who was intrested in implementing sound mode support for the sonos platform based on my code (but he wanted to wait until I finished and it has been quite a long time).

But that would already make 3 platforms that would uses this.
Besides almost all sound systems have some kind of sound mode feature, therefore it could be implemented in most of the platforms if someone is willing to make the code (and has that device).

Therfore I made this a general support feature in the media_player/__init__.py file, and I think this is a good thing. Do you agree @balloob?

Reply:
See https://developers.home-assistant.io/docs/en/entity_index.html#changing-the-entity-model

Reply:
So let me close this PR and the frontend PR, and you can open an issue on the architecture repo to propose the new feature.